win32ss/user/ntuser/class.c

```diff
@@ -1880,30 +1903,22 @@ UserSetClassLongPtr(IN PCLS Class,
 
     if (Index >= 0)
     {
-        PULONG_PTR Data;
-
         TRACE("SetClassLong(%d, %x)\n", Index, NewLong);
 
-        if (((ULONG)Index + sizeof(ULONG_PTR)) < (ULONG)Index ||
-            ((ULONG)Index + sizeof(ULONG_PTR)) > (ULONG)Class->cbclsExtra)
+        if (((ULONG)Index + Size) < (ULONG)Index ||
+            ((ULONG)Index + Size) > (ULONG)Class->cbclsExtra)
         {
             EngSetLastError(ERROR_INVALID_PARAMETER);
             return 0;
         }
 
-        Data = (PULONG_PTR)((ULONG_PTR)(Class + 1) + Index);
-
-        /* FIXME: Data might be a unaligned pointer! Might be a problem on
```

Comment:
Is this comment not relevant to how you are accessing the data in the new function?

Reply:
Good point, it does still apply. I will properly fix it for arm, etc.